src/tree/settings/CommonProperties.ts

```diff
@@ -30,14 +30,12 @@ export const sortByProperties: ITreePropertyInfo<CommonSortBy>[] = [
     { property: 'Label', description: localize('vscode-docker.tree.settings.label', 'Sort alphabetically by label') }
 ];
 
-export function getCommonPropertyValue(item: DockerObject, property: CommonProperty): string {
+export function getCommonPropertyValue(item: { createdAt?: Date, size?: number }, property: CommonProperty): string {
```

Comment:
It might be cleaner and less error prone to split this up into "getNormalizedCreatedTime" and "getNormalizedSize" or something along those lines, since this is effectively two different functions glued together with a switch statement.

Reply:
This is used by the grouping/labeling system in the tree views. Since the property name to look at (e.g. `CreatedTime` or `Size`) comes from VSCode settings, there would inevitably be a switch statement _somewhere_--either here, where the two "function bodies" are, or in a method reaching out to those two separated functions.

All of the rest of them use something like `getContainerPropertyValue` / `getImagePropertyValue` / etc. and contain all the "bodies" in big switch statements. We could of course refactor that but is there much value?

Reply:
Makes sense; I'd initially found this method via cases where it was being called with a hardcoded value for the property, but if it gets used dynamically as well, the potential benefit of a refactor is pretty minimal.